### ISSUE TITLE
Integrate the ReadOnly schema

### ISSUE THREAD

**Issue by hand-dot** [pdfme/pdfme#541]
### Is your feature request related to a problem? Please describe.

Until now, the ReadOnly schema and normals were separated into different schemas, but we will integrate them. We will make it possible to set ReadOnly from a checkbox, similar to how the 'required' checkbox is set.

- ref: https://github.com/pdfme/pdfme/pull/521#pullrequestreview-2207699785

### Describe the solution you'd like

- Add ReadOnly checkbox
  - ![image](https://github.com/user-attachments/assets/b7e46ca6-fe84-419f-befc-2d08fcbdf842)
- Integrate the ReadOnly schema

### Describe alternatives you've considered

None

### Additional context

This modification should be a major version update if we're going to delete the existing ReadOnly schema. It might be better to keep the ReadOnly schema for a while, but with this change, ReadOnly schema will eventually become unnecessary, so it's planned to be deprecated in the future.

**Comment by peteward** [comment]
I'm wondering if there's a better (English) word to use instead of "read only". I don't know if this would be clear to an editor.
Maybe **static** vs **dynamic** or **fixed** vs **editable**. But I guess this is just a label, and at the code level it's fine to be `readOnly`

**Comment by Mystificious** [comment]
> I'm wondering if there's a better (English) word to use instead of "read only". I don't know if this would be clear to an editor. Maybe **static** vs **dynamic** or **fixed** vs **editable**. But I guess this is just a translation label, and at the code level it's fine to be `readOnly`

Maybe "Not editable" ?

**Comment by hand-dot** [comment]
> Maybe static vs dynamic or fixed vs editable. 

> Maybe "Not editable" ?

I can't decide on my own. @peteward I want to trust your judgment.

**Comment by hand-dot** [comment]
If @peteward hasn't started working on it yet, would it be okay if I implement it myself?

**Comment by peteward** [comment]
Of course. 

Yes, I haven't started. 👍

**Comment by dkastl** [comment]
I actually did this in a project, where I extended some schemas with additonal fields. These additonal fields are not relevant for PDFme, but I felt adding a "switch" widget for setting the read-only flag was relatively easy:

https://github.com/gtt-project/redmine_print_templates/blob/next/src/plugins/extended/schemaUtils.ts#L47-L52

It would be nice if integrating the ReadOnly setting into PDFme. I actually asked the question once in Discord.
